### PR TITLE
Remove the secondary option (toggles sidebar) from section definitions

### PR DESCRIPTION
**Pull request by jsnajdr**
I got this idea while debugging #45150: do we really need to set `secondary: false` or `secondary: true` in section definitions, to specify whether the section's UI has a sidebar or not?

There's another, more straightforward way for the `<Layout>` component to figure out whether to render a sidebar or not: did the route handler chain set `context.secondary` before proceeding to render?

If a handler only sets `context.primary`, then we render layout without a sidebar. Only when `context.secondary` is set, too, do we toggle a sidebar enabled layout.

After I tried the idea out, it seems to work perfectly. The `secondary` option was indeed redundant.

There are many routes that want an exception from the section's default: all routes in a section have a sidebar, except one special one that doesn't. Such routes had to dispatch a `hideSidebar()` Redux action prior to rendering.

Removing the `secondary` option also improves the correctness of the rendering and navigation process. What do I mean? Imagine we are on the `/themes/example.blog` page, with a rendered sidebar. Now we click on a theme link and navigate to `/theme/twentytwenty/example.blog`. The `theme` section _doesn't_ have a sidebar. The sequence of events is:
- load and activate the `theme` section, setting `hasSidebar` to `false` in Redux
- at this time, the `themes` page is still rendered, so the Redux update triggers a rerender of the _old_ page (!)
- only now execute the `page( '/theme/:theme/:site' )` handler for the target page and render it -- using the new `hasSidebar` value.

In other words, the `secondary: false` from the new section leaks into the UI of the old section. That's weird. The navigation should ideally happen in one step.

**How to test:**
Sections that used `secondary: false` or disabled the sidebar for individual routes include:
- `jetpack-connect`
- `signup`
- `log-in` and also `oauth` login routes used only on Desktop
- `purchase-product` (new thing from Jetpack)
- `theme` (individual theme's showcase page)
- `mailing-lists` (page where you unsubscribe from wpcom emails)
- `reader-full-post`

@tyxla you know almost all areas of Calypso unlike anyone else I'd say. Can you help me test this?

**Todo:**
There's one guided tour (Simple Payments Emails) that relied on the `hasSidebar` Redux selector to detect UI change. I'll need to rewrite that code to use something else.

### CODE REVIEW

File: client/sections.js
Comment (tyxla) [inline client/sections.js:457]: Nice 👍

**Review by tyxla — COMMENTED** [review]
Great cleanup, Jarda!

I remember I took a quick look at removing `secondary` back when I was working on setting up the Jetpack Cloud infrastructure because it felt unnecessary indeed. It didn't seem to have any other implications, other than the visual one I pointed out above.

I think there are some tests we can cleanup from the `secondary` - the ones I noticed were document head selectors tests and UI selectors tests.

Other than that, I've left some comments here and there, but I think this is great and I doubt it's going to break anything. I've reviewed most of the more interesting cases it's touching, but happy to thoroughly test it once it's a published PR 😉

File: client/document/index.jsx
Comment (jsnajdr) [reply client/document/index.jsx:156]: You mean the `editor` custom placeholder? It didn't work anyway for a very long time (the div was empty and invisible) and these days, the Classic Editor is completely disabled and unreachable.

I can move the removal to another PR, but I think it's low risk enough to piggy-back on this one. I can create a separate commit for the change.

File: client/document/index.jsx
Comment (jsnajdr) [reply client/document/index.jsx:150]: Yes, I noticed the server-side usage only after creating the PR and seeing `test-server` coming back red 🙂 I don't think that the sidebar-enabled placeholder will be missed much, and it doesn't justify adding so much complexity.

Over the last year, I've seen too many exceptions where the UI layout doesn't correspond to the blue-masterbar-sidebar-content template and where the layout progression on load looks outright horrible:
- Signup
- customized login for Jetpack, Woo, Crowdsignal etc.
- Jetpack Connect (dark background for many pages)
- Gutenframe editor

Things also get ugly when you have a color scheme enabled and the masterbar and sidebar colors are very different from what the placeholder shows.

That's why I'm in favor of showing as generic a placeholder as possible. Ideally, a light-grey background with a centered WP logo.

File: client/layout/guided-tours/tours/simple-payments-email-tour/index.js
Comment (jsnajdr) [reply client/layout/guided-tours/tours/simple-payments-email-tour/index.js:27]: This guided tour isn't finished yet, see the todo section in the PR description. I'll need to rewrite the `when` condition to use some DOM observer that waits for the sidebar element to appear.

File: client/layout/index.jsx
Comment (jsnajdr) [reply client/layout/index.jsx:142]: Yes, and so far it seems to me that this condition works 100% of the time.

File: client/my-sites/controller.js
Comment (jsnajdr) [reply client/my-sites/controller.js:102]: Yeah, I added this code myself in #37816 😉 It's because when rendering the site selector or no-sites UI, we're rendering a UI that is not really related to the target section and shouldn't inherit, e.g., color styles from it. Or the sidebar setting. Some sections define CSS styles for `.is-section-whatever`, with color overrides, and we don't want to use them here.

File: client/sections.js
Comment (jsnajdr) [reply client/sections.js:70]: I believe I deleted `secondary` from _all_ sections? Or didn't I?

File: client/document/index.jsx
Comment (tyxla) [reply client/document/index.jsx:150]: Totally agree and am in favor of keeping a simpler placeholder too 👍

File: client/document/index.jsx
Comment (tyxla) [reply client/document/index.jsx:156]: No strong feelings for moving it to another PR if it's something that no one uses or sees these days. Feel free to keep the removal in this PR 😉

File: client/sections.js
Comment (tyxla) [reply client/sections.js:70]: I can still see it under some sections, namely `purchases`, `notification-settings`, and several more below.

File: client/layout/guided-tours/tours/simple-payments-email-tour/index.js
Comment (tyxla) [reply client/layout/guided-tours/tours/simple-payments-email-tour/index.js:27]: Gotcha 👍

File: client/layout/index.jsx
Comment (tyxla) [reply client/layout/index.jsx:142]: My quick and non-exhaustive testing has the same results. As I mentioned earlier, happy to test this more thoroughly once the PR is ready to review.

File: client/my-sites/controller.js
Comment (tyxla) [reply client/my-sites/controller.js:102]: Makes sense 👍

**Review by tyxla — COMMENTED** [review]
Looking great, Jarda! 

Let's finish this and I'm going to do some thorough testing when it's ready.

Let me know if you need anything else in the meantime.

File: client/document/index.jsx
Comment (jsnajdr) [reply client/document/index.jsx:156]: I verified a long time ago that the editor placeholder does nothing, but never submitted that small PR that removes it.

File: client/sections.js
Comment (jsnajdr) [reply client/sections.js:70]: Oh, something/someone must have distracted me and I forgot to remove all the `secondary` flags 🙂 Finishing the task in https://github.com/Automattic/wp-calypso/pull/45216/commits/9d2c4e17306cd4af23ac82a9605fabfca7a72636

File: client/layout/guided-tours/tours/simple-payments-email-tour/index.js
Comment (jsnajdr) [reply client/layout/guided-tours/tours/simple-payments-email-tour/index.js:27]: @tyxla After I started looking at this guided tour, I realized it can be removed completely 🙂 See #45286 for all the details. After that removal PR is merged, this one can be rebased and will be ready for review and deploy.

Comment (tyxla) [reply client/layout/guided-tours/tours/simple-payments-email-tour/index.js:27]: Sounds great! I've reviewed that PR to unblock this one.

**Comment by jsnajdr** [comment]
@tyxla The guided tour removal was merged, and after rebase, this PR is ready for final review.

**Review by tyxla — APPROVED** [review]
I've performed some exhaustive testing on all the specific instances this PR is touching, and everything is working great. Changes look good too. 👍 

I've just added a small cleanup commit that removes the `secondary` prop from some tests where it's completely redundant.

Really cool cleanup @jsnajdr! Feel free to ship it when the tests are green! 🚢

**Pull request merged**
